pkg/bff/models/v1/registration.go

```diff
@@ -181,9 +183,76 @@ func (r *RegistrationForm) ValidateLegalPerson() ValidationErrors {
 }
 
 // Validate only the fields in the contacts step.
-func (r *RegistrationForm) ValidateContacts() ValidationErrors {
-	// TODO: implement
-	return nil
+func (r *RegistrationForm) ValidateContacts() (v ValidationErrors) {
+	if r.Contacts == nil {
+		v = append(v, &ValidationError{Field: "contacts", Err: ErrMissingField.Error()})
+		return v
+	}
+
+	// Validate each non-zero contact
+	contacts := 0
+	iter := models.NewContactIterator(r.Contacts, false, false)
+	for iter.Next() {
+		contacts++
+		contact, field := iter.Value()
+		if !models.ContactIsZero(contact) {
+			v, _ = v.Append(ValidateContact(contact, "contacts."+field))
+		}
+	}
+
+	// Check that all required contacts are present (special rules)
+	switch contacts {
+	case 0:
+		// At least one contact is required
+		v = append(v, &ValidationError{Field: "contacts", Err: ErrNoContacts.Error()})
+	case 1:
+		// If there is only one contact, it must be the admin
+		if models.ContactIsZero(r.Contacts.Administrative) {
+			v = append(v, &ValidationError{Field: "contacts", Err: ErrMissingContact.Error()})
+		}
+	default:
+		// If there are at least two contacts, either admin or technical must be present
+		if models.ContactIsZero(r.Contacts.Administrative) && models.ContactIsZero(r.Contacts.Technical) {
+			v = append(v, &ValidationError{Field: "contacts", Err: ErrMissingContact.Error()})
+		}
+
+		// Admin or legal must be present
+		if models.ContactIsZero(r.Contacts.Administrative) && models.ContactIsZero(r.Contacts.Legal) {
+			v = append(v, &ValidationError{Field: "contacts", Err: ErrMissingContact.Error()})
+		}
+	}
```

Comment:
```suggestion
		// Check that all required contacts are present (special rules)
	switch contacts {
	case 0:
		// At least one contact is required
		v = append(v, &ValidationError{Field: "contacts", Err: ErrNoContacts.Error()})
	case 1:
		// If there is only one contact, it must be the admin; if not highlight the missing fields
		if models.ContactIsZero(r.Contacts.Administrative) {
			// Global contact error
			v = append(v, &ValidationError{Field: "contacts", Err: ErrMissingContact.Error()})

			switch {
			case !models.ContactIsZero(r.Contacts.Technical):
				// If the technical contact is filled in then nominate the legal/admin contact to be populated
				v = append(v, &ValidationError{Field: "contacts.administrative", Err: "specify either the administrative or legal contact"})
				v = append(v, &ValidationError{Field: "contacts.legal", Err: "specify either the administrative or legal contact"})
			case !models.ContactIsZero(r.Contacts.Legal):
				// If the legal contact is filled in then nominate the technical/admin contact to be populated
				v = append(v, &ValidationError{Field: "contacts.administrative", Err: "specify either the administrative or technical contact"})
				v = append(v, &ValidationError{Field: "contacts.technical", Err: "specify either the administrative or technical contact"})
			default:
				// Otherwise say that one of the other fields is required
				v = append(v, &ValidationError{Field: "contacts.administrative", Err: ErrMissingContact.Error()})
				v = append(v, &ValidationError{Field: "contacts.technical", Err: ErrMissingContact.Error()})
				v = append(v, &ValidationError{Field: "contacts.legal", Err: ErrMissingContact.Error()})
			}
		}
	default:
		// If there are at least two contacts, either admin or technical must be present
		if models.ContactIsZero(r.Contacts.Administrative) && models.ContactIsZero(r.Contacts.Technical) {
			v = append(v, &ValidationError{Field: "contacts", Err: ErrMissingContact.Error()})
			v = append(v, &ValidationError{Field: "contacts.administrative", Err: "specify either the administrative or technical contact"})
			v = append(v, &ValidationError{Field: "contacts.technical", Err: "specify either the administrative or technical contact"})
		}

		// Admin or legal must be present
		if models.ContactIsZero(r.Contacts.Administrative) && models.ContactIsZero(r.Contacts.Legal) {
			v = append(v, &ValidationError{Field: "contacts", Err: ErrMissingContact.Error()})
			v = append(v, &ValidationError{Field: "contacts.administrative", Err: "specify either the administrative or legal contact"})
			v = append(v, &ValidationError{Field: "contacts.legal", Err: "specify either the administrative or legal contact"})
		}
	}
```